# **Confidence Intervals**

In [32]:
# import necessary libraries

import numpy as np # for linear algebra
import pandas as pd # for creating dataframes

from statsmodels.formula.api import ols # for Ordinary Least Squares method

In [33]:
# import the dataset

sleep_study_data = pd.read_csv('sleep75.csv')
sleep_study_data.shape

(706, 34)

## Estimate the regression

In [34]:
we_need_to_sleeeeeep = ols(formula='sleep~totwrk+age+south+male+smsa+marr+union', data=sleep_study_data).fit()
we_need_to_sleeeeeep.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                  sleep   R-squared:                       0.131
Model:                            OLS   Adj. R-squared:                  0.123
Method:                 Least Squares   F-statistic:                     15.07
Date:                Wed, 13 May 2026   Prob (F-statistic):           2.11e-18
Time:                        12:46:44   Log-Likelihood:                -5255.9
No. Observations:                 706   AIC:                         1.053e+04
Df Residuals:                     698   BIC:                         1.056e+04
Df Model:                           7                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept   3441.2975     78.942     43.593      0.000    3286.305    3596.290
totwrk        -0.1689      0.018     -9.363      0.000      -0.204      -0.133
age            2.8413      1.387      2.048      0.041       0.117       5.565
south        101.6755     41.834      2.430      0.015      19.541     183.810
male          86.0381     34.906      2.465      0.014      17.505     154.571
smsa         -53.9249     33.156     -1.626      0.104    -119.022      11.172
marr          30.0061     41.910      0.716      0.474     -52.278     112.290
union         12.4396     38.096      0.327      0.744     -62.357      87.236
==============================================================================
Omnibus:                       61.619   Durbin-Watson:                   1.946
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              164.554
Skew:                          -0.444   Prob(JB):                     1.85e-36
Kurtosis:                       5.192   Cond. No.                     1.23e+04
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 1.23e+04. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

In [35]:
we_need_to_sleeeeeep.params.round(3)

Intercept    3441.297
totwrk         -0.169
age             2.841
south         101.676
male           86.038
smsa          -53.925
marr           30.006
union          12.440
dtype: float64

## Calculate the critical value

In [36]:
# Option 1
# significance_level (α)= 0.05 = 5%
# degrees_of_freedom = n-k-1 = 706-7-1 (for this specific case)
# n - number of observations
# k - number of regressors
# t.ppf
t.ppf(q=1-0.05/2,df=706-7-1).round(3)

1.963

In [37]:
# Option 2
# t.isf
t.isf(q=0.05/2,df=706-7-1).round(3)

1.963

## Calculate the confidence intervals for each coefficient separately

The confidence interval helps us to determine statistical significance of an individual regressor at a certain significance level.

If the confidence interval **includes zero** we fail to reject H0.<br>
If the confidence interval **excludes zero** we reject H0.

Usually H0: β = 0

Confidence intervals are intervals first of all. So, they include two points.<br>
A confidence interval can be interpreted as 'with α% certainty the effect of the X for the Y lies within the interval \[...\]'.

I = (β - Std_err \* t_cr; β + Std_err \* t_cr)<br>
The lower boundary: β-Std_err\*t_cr<br>
The upper boundary: β+Std_err\*t_cr

**Examples for α = 5%:**<br>
A 95% confidence interval that excludes zero ⇔ Reject H0: βj = 0 at  α = 0.05<br>
A 95% confidence interval that includes zero ⇔ Fail to reject H0: βj = 0 at α = 0.05

### Using the built-in method for all regressors at once

In [38]:
we_need_to_sleeeeeep.conf_int(alpha=0.05)

,0,1
Intercept,3286.305108,3596.289861
totwrk,-0.204293,-0.133470
age,0.117192,5.565350
south,19.540715,183.810336
male,17.505333,154.570894
smsa,-119.021680,11.171884
marr,-52.277772,112.289955
union,-62.356615,87.235812


### Manually for individual resgressors

#### totwrk

In [39]:
# β - Std_err * t_cr
boundary_1=-0.169-0.018*1.963
boundary_1

-0.20433400000000002

In [40]:
# β + Std_err * t_cr
boundary_2=-0.169+0.018*1.963
boundary_2

-0.133666

Answer: **totwrk** confidence interval does not include a zero; **totwrk** is statistically significant. With 95% certainty the effect of the **totwrk** on **sleep** lies within the interval \[-0.2043; -0.1337\]

#### south

In [41]:
# β - Std_err * t_cr
boundary_1=101.6755-41.834*1.963
boundary_1

19.555357999999984

In [42]:
# β + Std_err * t_cr
boundary_2=101.6755+41.834*1.963
boundary_2

183.79564200000002

Answer: **south** confidence interval does not include a zero; **south** is statistically significant. With 95% certainty the effect of the **south** on **sleep** lies within the interval \[19.5554; 183.7956\]

#### smsa

In [43]:
# β - Std_err * t_cr
boundary_1=-53.925-33.156*1.963
boundary_1

-119.010228

In [44]:
# β + Std_err * t_cr
boundary_2=-53.925+33.156*1.963
boundary_2

11.160228000000004

Answer: **smsa** confidence interval includes a zero; **smsa** is statistically insignificant.